ladybug_comfort/collection/solarcal.py

```diff
@@ -0,0 +1,530 @@
+# coding=utf-8
+"""Objects for calculating solar-adjusted MRT from DataCollections."""
+from __future__ import division
+
+from ..solarcal import outdoor_sky_heat_exch, indoor_sky_heat_exch, \
+    shortwave_from_horiz_solar, sharp_from_solar_and_body_azimuth
+from ..parameter.solarcal import SolarCalParameter
+from ._base import ComfortCollection
+
+from ladybug.location import Location
+from ladybug.sunpath import Sunpath
+from ladybug.datacollection import HourlyDiscontinuousCollection
+
+from ladybug.datatype.temperature import Temperature, MeanRadiantTemperature
+from ladybug.datatype.temperaturedelta import RadiantTemperatureDelta
+from ladybug.datatype.energyflux import Irradiance, EffectiveRadiantField, \
+    HorizontalInfraredRadiationIntensity
+from ladybug.datatype.energyintensity import Radiation
+from ladybug.datatype.fraction import Fraction
+
+
+class _SolarCalBase(ComfortCollection):
+    """Base class used by all objects that use SolarCal with Data Collections."""
+
+    @property
+    def location(self):
+        """Ladybug Location object."""
+        return self._location.duplicate()
```

Comment:
Again, if the ladybug location were immutable, I would not have to duplicate this.  For now, I have to do this to avoid the location on the solarcal object from getting out of sync with the results.

Reply:
Replied up there. I mark this one as resolved.

Reply:
Wait, but this makes it unclear if we also need to add a `read_only `attribute to the Ladybug Location object.  I can see why we maybe want to do this for the DataCollection but, after realizing that we would also need to do this on the Location object, there's an apparent slippery slope to adding read_only attributes on all Ladybug objects that have setters.  I am not sure that we want to do this and I want to hear your thoughts here.

Reply:
We don't need to. By replying above I was referring to overwrite the `__setattr__` in this class itself. In this case and since location is a very light object creating a new instance at each call should be fine. I don't think we want to make this one an immutable object.

Reply:
Ok. I agree that it would be weird to have the location object as immutable and I am glad to hear that you think that the object is small enough that we could just duplicate it for this one case.  Speaking of this, I found a bug in how I implemented the Location.duplicate() method and we will have to merge in [this pul request](https://github.com/ladybug-tools/ladybug/pull/120) before we merge in this comfort one.